Fix test keys

lukpueh opened the pull request:
See detailed commit messages for more information.
Note: We should think about regression tests at some point.

Commit:
Homogenize testing demo keys

The meta data files in 'test/demo_files' were signed by keys
that are not longer in the repository. Neither the key pairs
for alice, bob and carl in 'test/demo_files', nor the key pairs
for alice, bob and carl in the demo directory were used.
Moreover the testing key pairs and the demo key pairs do not
correspond.

While this does not break unittesting, which uses the keys and
the meta data files independently. It is confusing and leads to
problems if the existing test meta data needs to be resigned,
e.g. because of changes in the meta data format.

This commit, as first part of a fix, homogenizes the key pairs
in test/demo_files and the demo directory for alice, bob and
carl respectively

Commit:
Re-sign/rename test link meta data

Re-sign/re-name test link meta data and adapt unit tests
to use new keys (see previous commit).

The following shell snippet was used to re-sign/re-name
the link files.

```shell
in-toto-sign sign -kbob -ir write-code.c8650e01.link
in-toto-sign sign -kcarl -ir package.2dc02526.link
```

Commit:
Adapt test layout template to list keys

Used below snippet to add the updated public keys
to the testing layout's keystore and step authorization.

Note: Manually removed expiration date to prevent unit tests
to fail due to expiration

```python
from in_toto import util
from in_toto.models import layout

bob_pub = util.import_rsa_key_from_file("bob.pub")
carl_pub = util.import_rsa_key_from_file("carl.pub")
layout_obj = layout.Layout.read_from_file("demo.layout.template")
layout_obj.keys = {
  bob_pub["keyid"]: bob_pub,
  carl_pub["keyid"]: carl_pub
}
layout_obj.steps[0].pubkeys = [bob_pub["keyid"]]
layout_obj.steps[1].pubkeys = [carl_pub["keyid"]]
layout_obj.dump("demo.layout.template")
```

SantiagoTorres commented:
>  Note: We should think about regression tests at some point.

👍

SantiagoTorres reviewed (approved):
Good commit descriptions and conceptual separation. Thanks!

Commit:
Add expected_command shlex parsing unit test

The previous commit modified the layout template using in-toto
tooling. As consequence a test scenario, i.e. parsing a command
String into a list, disappeared.
The newly added unit test restores the testing coverage.
